spec/controllers/reservations_controller_spec.rb

```diff
@@ -1,247 +1,189 @@
-      before(:each) { get :show, id: @reservation.id }
-      it { is_expected.to render_template(:show) }
-      it { expect(response).to be_success }
-      it { expect(assigns(:reservation)).to eq @reservation }
-    end
-
-    shared_examples 'can view reservation by admin' do
-      before(:each) { get :show, id: @admin_res.id }
-      it { is_expected.to render_template(:show) }
-      it { expect(response).to be_success }
-      it { expect(assigns(:reservation)).to eq @admin_res }
     end
 
-    shared_examples 'cannot view reservation by admin' do
-      before(:each) { get :show, id: @admin_res.id }
-      include_examples 'cannot access page'
-    end
-
-    context 'when accessed by admin' do
-      before(:each) do
-        sign_in @admin
+    context 'normal user' do
+      let!(:user) do
+        res = spy('Array', starts_on_days: time_filtered)
+        UserMock.new(reservations: res)
       end
-
-      it_behaves_like 'can view reservation by patron'
-      it_behaves_like 'can view reservation by admin'
-    end
-
-    context 'when accessed by checkout person' do
       before(:each) do
-        sign_in @checkout_person
+        mock_user_sign_in(user)
+        get :index
       end
-
-      it_behaves_like 'can view reservation by patron'
-      it_behaves_like 'can view reservation by admin'
+      it { is_expected.to respond_with(:success) }
+      it { is_expected.to render_template(:index) }
+      it "only gets the current user's reservations" do
+        expect(user).to have_received(:reservations).at_least(:once)
+      end
+      it 'defaults to reserved' do
+        # twice: once from set_counts, once from filtering
+        expect(time_filtered).to have_received(:reserved).twice
+      end
+      it_behaves_like 'filterable'
     end
 
-    context 'when accessed by patron' do
+    context 'admin' do
+      let!(:user) { UserMock.new(:admin) }
       before(:each) do
-        sign_in @user
+        allow(Reservation).to receive(:starts_on_days)
+          .and_return(time_filtered)
+        mock_user_sign_in(user)
+        get :index
       end
-
-      it_behaves_like 'can view reservation by patron'
-      it_behaves_like 'cannot view reservation by admin'
+      it 'gets all reservations' do
+        expect(Reservation).to have_received(:starts_on_days)
+      end
+      it 'defaults to upcoming' do
+        # twice: once from set_counts, once from filtering
+        expect(time_filtered).to have_received(:upcoming).twice
+      end
+      it_behaves_like 'filterable'
     end
 
     it_behaves_like 'inaccessible by banned user' do
-      before { get :show, id: @reservation.id }
+      before { get :index }
     end
   end
 
   describe '#new (GET /reservations/new)' do
-    # unhappy paths: banned user, there is no reservation in the cart
+    # SMELLS:
+    #   - long method
+    #   - redirect location set for testing in code ???
+    #   - feature envy
+    # most of this should probably be in the cart?
```

Comment:
Oh whoops this is `new`, not `create`. Yea, we could probably go either way with Service Object vs, putting it in `Cart`.